syntax = "proto3";

option go_package = "./ecommerce";
// option go_package = "github.com/8bit-euclid/grpc-sandbox/ecommerce";
// option go_package = "github.com/8bit-euclid/grpc-sandbox/gen/go/ecommerce";
package ecommerce;

service ProductInfo {
    rpc addProduct(Product) returns (ProductID);
    rpc getProduct(ProductID) returns (Product);
}

message Product {
    string id = 1;
    string name = 2;
    string description = 3;
    float price = 4;
}

message ProductID {
    string value = 1;
}